lightning/src/routing/router.rs

```diff
@@ -937,42 +946,55 @@ impl cmp::PartialOrd for RouteGraphNode {
 ///
 /// Used to construct a [`PathBuildingHop`] and to estimate [`EffectiveCapacity`].
 #[derive(Clone, Debug)]
-enum CandidateRouteHop<'a> {
+pub enum CandidateRouteHop<'a> {
 	/// A hop from the payer, where the outbound liquidity is known.
 	FirstHop {
+		/// The channel details of the first hop.
 		details: &'a ChannelDetails,
+		/// The node id of the payer.
+		node_id: NodeId
 	},
 	/// A hop found in the [`ReadOnlyNetworkGraph`], where the channel capacity may be unknown.
 	PublicHop {
+		/// The channel info of the hop.
 		info: DirectedChannelInfo<'a>,
+		/// The short_channel_id of the channel.
 		short_channel_id: u64,
 	},
 	/// A hop to the payee found in the BOLT 11 payment invoice, though not necessarily a direct
 	/// channel.
 	PrivateHop {
+		/// Hint about the the hop
 		hint: &'a RouteHintHop,
+		/// The node id of the payee.
+		target_node_id: NodeId
```

Comment:
so started doing this and wonder about this `unwrap` (L1213). how should this be tackled?
another q, should I leave the bool input(source < target) in the `id` function as an input or change it because 
now we have `source` and `target` functions in `CandidateRouteHop`
```rust
     4 impl<'a> core::fmt::Debug for PathBuildingHop<'a> {
     3         fn fmt(&self, f: &mut core::fmt::Formatter) -> Result<(), core::fmt::Error> {
     2                 let mut debug_struct = f.debug_struct("PathBuildingHop");
     1                 debug_struct
| 1213                         .field("node_id", &self.candidate.id(&self.candidate.source() < &self.candidate.target().unwrap()))
     1                         .field("short_channel_id", &self.candidate.short_channel_id())
     2                         .field("total_fee_msat", &self.total_fee_msat)
     3                         .field("next_hops_fee_msat", &self.next_hops_fee_msat)
     4                         .field("hop_use_fee_msat", &self.hop_use_fee_msat)
     5                         .field("total_fee_msat - (next_hops_fee_msat + hop_use_fee_msat)", &(&self.total_fee_msat - (&self.next_hops_fee_msat + &sel
       f.hop_use_fee_msat)))
     6                         .field("path_penalty_msat", &self.path_penalty_msat)
     7                         .field("path_htlc_minimum_msat", &self.path_htlc_minimum_msat)
     8                         .field("cltv_expiry_delta", &self.candidate.cltv_expiry_delta());
     9                 #[cfg(all(not(ldk_bench), any(test, fuzzing)))]
    10                 let debug_struct = debug_struct
    11                         .field("value_contribution_msat", &self.value_contribution_msat);
    12                 debug_struct.finish()
    13         }
    14 }
    ```